### ISSUE TITLE
renamed, deleted and removed delayed deleted mailbox leave mails and folders in archive partition

### ISSUE THREAD

**Issue by MichaelMenge** [cyrusimap/cyrus-imapd#2657]
I have discovered that for some deleted accounts there where still directories and mails for this
account on the archive partition. 

I am using cyrus 3.0.8, with expunge_mode: delayed, delete_mode: delayed, 
metapartition and archivepartition enabled. Still investigating if this happens always 
or only sometimes

**Comment by slangenhorst** [comment]
Same here with 3.2.6, expunge_mode: delayed, delete_mode: delayed
Just removed "manually" 3TB of old files and dirs from the archive-partition.

**Comment by MichaelMenge** [comment]
```imap/mailbox.c:HIDDEN int mailbox_delete_cleanup``` does not handle archivepartitions so, any renamed, deleted and removed delayed deleted mailbox will leave mails and folders in the archivepartition. 

As far as i can tell this bug is present  since the archivepartition options implemented an is also in 3.4 and master

**Comment by brong** [comment]
Yep!  That makes sense.  I'm working in this area already, so I'll make sure it's fixed on master, and backport fixes to earlier versions too.

**Comment by brong** [comment]
This should be fixed in all supported versions now.
